### PR TITLE
ASU-565 Add REST API for user/profile CRUD operations

### PR DESCRIPTION
**Pull request by jmp**
This PR adds the ability to create, retrieve, update, and delete profiles and users via a REST API.

The API does not yet require authentication via JWT. This will be added in a separate PR.

### CODE REVIEW

**Review by tomijarvi — APPROVED** [review]
Clean implementing. Good work! 💯

File: users/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                "first_name": "Maija",
+                "address": "Kauppakatu 23",
+            }
+            APIClient().put(url, put_data)
+    profile.refresh_from_db()
+    assert profile.user.first_name == PROFILE_TEST_DATA["first_name"]
+    assert profile.address == PROFILE_TEST_DATA["address"]
+
+
+def _create_profile_from_data(profile_data: dict) -> Profile:
```
Comment (genie9) [inline users/tests/test_api.py:129]: Should this be better as a fixture?

Comment (jmp) [reply users/tests/test_api.py]: Modified to a fixture. 👍 It actually made some of the tests a bit cleaner.

File: users/api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from rest_framework.fields import CharField, EmailField
+from rest_framework.serializers import ModelSerializer
+
+from ..models import Profile
+
```
[the remark this answers is not in the document]
Comment (jmp) [reply users/api/serializers.py:7]: Changed to absolute imports.

**Review by genie9 — APPROVED** [review]

**Pull request merged**